crates/oxc_parser/src/lexer/number.rs

```diff
@@ -120,5 +124,193 @@ pub fn parse_big_int(s: &str, kind: Kind) -> Result<BigInt, &'static str> {
         Kind::Hex => 16,
         _ => unreachable!(),
     };
-    BigInt::parse_bytes(s.as_bytes(), radix).ok_or("invalid bigint")
+    // NOTE: BigInt::from_bytes does a utf8 check, then uses from_str_radix
+    // under the hood. We already have a string, so we can just use that
+    // directly.
+    BigInt::from_str_radix(s, radix).map_err(|_| "invalid bigint")
+}
+
+#[cfg(test)]
+#[allow(clippy::unreadable_literal, clippy::mixed_case_hex_literals)]
+mod test {
+
+    use super::{parse_float, parse_int, Kind};
+
+    /// Assert that all (&str, number) items in an [`Iterator`] parse to the
+    /// expected number. `number` will be casted to an [`f64`].
+    macro_rules! assert_all_eq {
```

Comment:
Do we really need macros instead of simple functions 😅 . This file is at the bottom of the dependency tree, it is valuable to save as much compilation time as possbile.

Reply:
I would also prefer to lose the macros here, but not for compile time reasons.

I feel like tests should be as "stupid" as possible so it's really clear exactly what they test and to verify that they're testing what you think they are.

In my view, having a pretty simple macro to shorten a repeated pattern is fine, but this macro is too clever! (by which I mean it takes brain power to understand)

To remove the macro while also avoiding repetition, could you make `assert_all_eq!` (or at least parts of it) into a normal function?

Reply:
Removed